examples/async_pcb_insert_drq/record_demo.py

```diff
@@ -53,13 +54,28 @@
         obs = next_obs
 
         if done:
-            print(rew)
             success_count += rew
+            total_count += 1
+            print(
+                f"{rew}\tGot {success_count} successes of {total_count} trials. {success_needed} successes needed."
+            )
             pbar.update(rew)
             obs, _ = env.reset()
 
     uuid = datetime.datetime.now().strftime("%Y-%m-%d_%H-%M-%S")
-    file_name = f"./bc_demos/pcb_insert_{success_needed}_demos_{uuid}.pkl"
-    with open(file_name, "wb") as f:
-        pkl.dump(transitions, f)
-        print(f"saved {success_needed} demos to {file_name}")
+    file_name = f"./pcb_insert_{success_needed}_demos_{uuid}.pkl"
```

Comment:
Is it easier if we check the dir exists beforehand? else create the dir if it doesnt exist. something like
```py
    if os.path.exists(DATA_DIRECTORY):
        shutil.rmtree(DATA_DIRECTORY)
    os.makedirs(DATA_DIRECTORY)
```

Reply:
Sure. We need to check if the dir exists, and whether we have write permission to that dir. We can start recording only after this check completes. Is that a preferable solution?

Also, I don't think we should do an `shutil.rmtree` on any users machine. The user might be storing demo files in the same directory as other code or files.

Reply:
Yes, just create the dir is suffice and maybe check the write permission too.

ahh yes, you are right, we shouldnt do `rmtree`, I copied this code snippet from another codebase.

Reply:
added this logic